### ISSUE TITLE
Setting RAILS_ENV=test

### ISSUE THREAD

**Issue by danielpclark** [vim-test/vim-test#19]
I've tried setting the environment variable through vim-test configuration but it doesn't get appended in execution as a command.

When mapping with :Minitest [args] I get the following command output which results in an error.

```
bundle exec rake test TEST="test/**/*_test.rb" TESTOPTS="RAILS_ENV=test"
```

Also trying either of these two options don't work either.

```
let g:test#ruby#minitest#options = 'RAILS_ENV=test'
let g:test#ruby#minitest#options = {'RAILS_ENV': '=test'}
```

**Comment by janko-m** [comment]
Yes, I see what you mean. Ideally the `RAILS_ENV=test` should be *prepended* to the test command.

I could theoretically extend the API in the way that you can do something like

```
let g:test#ruby#minitest#environment = {'RAILS_ENV': 'test'}
```

However, I want to do it only if it's necessary. I was always able to avoid setting environment variables for common tests. Did you try adding `ENV["RAILS_ENV"] = "test"` in the top of your `test/test_helper.rb`

I'm aware that sometimes you need to set some environment variables for tests, e.g. `rake test COVERAGE=1`, but I think commands like these should be executed manually (test.vim is only for regular tests).

**Comment by danielpclark** [comment]
I haven't put that in the `test_helper.rb`.  I guess if I'm planning on hardcoding the environment like that I could put it in my `.bashrc`.  So it's not super important for me to have it with vim-test.  It might be nice to have as an option though.

Feel free to close the issue.  When I try to do things one way I tend to forget that there are other ways of performing the same task.  Thanks for the tip!

**Comment by janko-m** [comment]
`ENV["RAILS_ENV"] = "test` is actually autogenerated by all major testing frameworks ([rspec-rails does it](https://github.com/rspec/rspec-rails/blob/master/lib/generators/rspec/install/templates/spec/rails_helper.rb#L2), [minitest-rails does it](https://github.com/blowmage/minitest-rails/blob/master/lib/generators/minitest/install/templates/test_helper.rb#L1)). This is much better than having to do prepend each test command with `RAILS_ENV=test`.

Setting `RAILS_ENV=test` in `.bashrc` wouldn't be a right thing to do, because then all of your commands would by default be run in "test" environment, including `rails server/console` and `rake db:*` tasks. `RAILS_ENV=test` should only be set when running tests (obviously :smiley:), so that's why the best place is the test helper.

**Comment by danielpclark** [comment]
Ah.  Thanks again!

**Comment by cb-bffmike** [comment]
I would love if this issue was reopened. I am testing typescript with jest, and specifically with date testing, I set the time zone to UTC: "TZ=UTC jest ...".

**Comment by codeinabox** [comment]
> I would love if this issue was reopened. I am testing typescript with jest, and specifically with date testing, I set the time zone to UTC: "TZ=UTC jest ...".

In this instance is setting the environment variable in the test itself an option?

**Comment by codeinabox** [comment]
This has since been resolved with the commit aff3607 that allows you to pass the environment variable eg `:Minitest RAILS_ENV=test`
